### ISSUE TITLE
how to plot confusion matrix with tf.data.Dataset?

### ISSUE THREAD

**Issue by caesariodito** [caesariodito/k-thesis#8]
separate the tf.data.Dataset variables into images and labels
then do predictions and plot the confusion matrix

I've done the steps above but I still encounter error

**Comment by caesariodito** [comment]
another option is to load only the X_test and y_test using the custom function, but I think it's a bit unefficient

![image](https://github.com/caesariodito/k-thesis/assets/75000641/8ca4e698-f42c-42fc-96ad-7412550abad9)

**Comment by caesariodito** [comment]
nvm I can use this code:

```python
y_pred = []  # store predicted labels
y_true = []  # store true labels

# iterate over the dataset
for image_batch, label_batch in test_ds_mapped:   # use dataset.unbatch() with repeat
   # append true labels
   y_true.append(label_batch)
   # compute predictions
   preds = vgg_transferred_half.predict(image_batch)
   # append predicted labels
   y_pred.append(np.argmax(preds, axis = - 1))

# convert the true and predicted labels into tensors
correct_labels = tf.concat([item for item in y_true], axis = 0)
predicted_labels = tf.concat([item for item in y_pred], axis = 0)
```

```python
from sklearn.metrics import confusion_matrix
import seaborn as sns

# Make predictions
# predictions = vgg_transferred_half.predict(X_test)
# y_pred = predictions.argmax(axis=1)

cm = confusion_matrix(correct_labels, predicted_labels)

plt.figure(figsize=(20, 20))
sns.heatmap(cm, annot=True, fmt='d', cmap='Blues', xticklabels=class_names, yticklabels=class_names)
plt.xlabel('Predicted')
plt.ylabel('True')
plt.show()
```

ss:
![image](https://github.com/caesariodito/k-thesis/assets/75000641/fbd494bf-2230-414f-b90a-d3138a0d8f8e)

thanks to: https://stackoverflow.com/questions/64622210/how-to-extract-classes-from-prefetched-dataset-in-tensorflow-for-confusion-matri
